kratos/solving_strategies/schemes/residualbased_incrementalupdate_static_scheme.h

```diff
@@ -192,20 +192,33 @@ class ResidualBasedIncrementalUpdateStaticScheme
 
         ProcessInfo& r_current_process_info = rModelPart.GetProcessInfo();
 
-        // Initialize solution step for all of the elements
+        // Definition of the first element iterator
+        const auto it_elem_begin = rModelPart.ElementsBegin();
+
         #pragma omp parallel for
-        for(int i=0; i<static_cast<int>(rModelPart.Elements().size()); i++) {
-            auto it_elem = rModelPart.ElementsBegin() + i;
+        for(int i=0; i<static_cast<int>(rModelPart.Elements().size()); ++i) {
+            auto it_elem = it_elem_begin + i;
             it_elem->InitializeNonLinearIteration(r_current_process_info);
         }
 
-        // Initialize solution step for all of the conditions
+        // Definition of the first condition iterator
+        const auto it_cond_begin = rModelPart.ConditionsBegin();
+
         #pragma omp parallel for
-        for(int i=0; i<static_cast<int>(rModelPart.Conditions().size()); i++) {
-            auto it_cond = rModelPart.ConditionsBegin() + i;
+        for(int i=0; i<static_cast<int>(rModelPart.Conditions().size()); ++i) {
+            auto it_cond = it_cond_begin + i;
             it_cond->InitializeNonLinearIteration(r_current_process_info);
```

Comment:
why not
```suggestion
            (it_cond_begin + i)->InitializeNonLinearIteration(r_current_process_info);
```
?

Reply:
Well, we can do that, but it is not so relevant (and taking into account that for the Release 7.0 we must cherrypick the less commits the better)